### PR TITLE
traffic_reports: add lines in the tf section

### PR DESCRIPTION
**Pull request by vpassama**
Hello,

This is a small PR where we add line objects directly in traffic_report API even for line_section disruptions.

Requesting impacted line by line_section `/lines/line:id/traffic_reports` will now display the impacted line in traffic_report.

There is a small change for tests about stop_point / stop_area objects as the disruption will now appear even if the object is not directly impacted by the line_section. The link is still on the line object and not on the stop_point / stop_area in this case.

Thank you.

### CODE REVIEW

**Comment by buildbotctp** [comment]
Can one of the admins verify this patch?

**Comment by kinnou02** [comment]
Jenkins, test this please.

**Review by kinnou02 — APPROVED** [review]

**Review by nonifier — APPROVED** [review]

File: source/jormungandr/tests/line_sections_tests.py
Comment (pbougue) [inline source/jormungandr/tests/line_sections_tests.py]: minor: has_`tr`_disruption looks better to me

File: source/jormungandr/tests/line_sections_tests.py
Comment (pbougue) [inline source/jormungandr/tests/line_sections_tests.py]: in traffic_reports' response to the request provided

File: source/jormungandr/tests/line_sections_tests.py
Comment (pbougue) [inline source/jormungandr/tests/line_sections_tests.py]: this seems difficult for integrator that want to filter and print only disruptions affecting stop_point/area

**Review by pbougue — COMMENTED** [review]
OK, so the way I understand it, when we only have line_section's disruptions, we used to have:
```json
"traffic_reports": [
  {
    "network": {...},
    "stop_areas": [...]
  }
]
```

And now we will have lines:
```json
"traffic_reports": [
  {
    "network": {...},
    "stop_areas": [...],
    "lines": [...]
  }
]
```
Am I correct?

I guess the behavior is pretty much the same on line_reports, so we have to pay attention I think (at least it should probably match).

Also, I think we should update the RFC on that if the functional of this PR doesn't change, especially on the part of stop_areas out of line_section: https://github.com/CanalTP/navitia/blame/dev/documentation/rfc/line_sections.md#L235 .

But I'm not super-fan of having that info on stops that are not affected by line_section disruptions... Shouldn't we filter a bit more on stops that are in the line_section? poke @TeXitoi 

@vpassama if the "strict" functional need you have is on `coverage/tisseo/line/line1/traffic_reports` and `coverage/tisseo/traffic_reports` then it's OK, and I think your feature is nice. But I vote it would be nicer to filter more when requesting on stops (don't know if it's so hard).

File: source/jormungandr/tests/line_sections_tests.py
Comment (pbougue) [reply source/jormungandr/tests/line_sections_tests.py]: looks like I'm the only one liking more `tr`, so nevermind :smiley:

**Comment by vpassama** [comment]
Hi @pbougue, thank you for the review.

You're right on what is expected in this PR, we want the field `lines` to be displayed for line_sections disruptions. The need is only on `lines/line/traffic_reports` and `/traffic_reports`.

Do you need to discuss more about the way you want to filter the stop_areas for an other PR or should I do it directly in this one ?

**Comment by pbougue** [comment]
So, as discussed on the phone:
As we already had the disruptions we wanted to output, let's just try to parse them after we selected them from stops.
Then from this parsing, add lines if they are line_section disruptions.

If the code is too complicated or dirty, we will try to have a look at it (and we will handle ourself the "cleanness" we want).

The final (but probably acceptable anyway) option would be to let the code as is (so don't loose it, please in case we want to have a look at it :mag: ) and just update the RFC.

File: source/jormungandr/tests/line_sections_tests.py
Comment (xlqian) [inline source/jormungandr/tests/line_sections_tests.py]: return any(disruption == link['id'] for link in pt_obj.get('links', []))

**Comment by nonifier** [comment]
@vpassama, @pbougue : Hey guys, just to follow-up on this PR as it's been here for quite some time... What was the outcome? are they any actions undertaken? How do we progress from here ? :)

**Comment by buildbotctp** [comment]
Can one of the admins verify this patch?

**Comment by mbounabi** [comment]
+1 regarding @nonifier comment: @vpassama, @pbougue, what are the next steps regarding this PR?

**Comment by vpassama** [comment]
Sorry I did not make any progress on this PR since the discussion I had with @pbougue (holidays + other work to do). So, as he said, we discussed about this and he told me to try some modifications with the code. I think I'll do it next week and I'll let you know if it works.

**Comment by vpassama** [comment]
Hi guys, @pbougue I made the modifications (quick & dirty), it seems all the tests pass. You said you would like a dedicated function maybe, I made this as a first step and if you want something better don't hesitate to tell me.

I used two functions to test the disruptions in traffic_reports, the first is like the old one, it only checks the disruption is in the response. The second is like the one I made in my first commit, it also checks that the specified object is linked to the disruption in the traffic_reports section of the response.

Doing so, I saw that the `/lines/line:2/traffic_reports` is linked to a disruption but the [RFC](https://github.com/CanalTP/navitia/blob/dev/documentation/rfc/line_sections.md#ptref-with-traffic_reports) says it shouldn't. Anyway, this happens without this PR.

**Comment by nonifier** [comment]
@pbougue : in all likelihood, you are in the best position to merge this PR if you think there is no more work to be done :)

File: source/jormungandr/tests/line_sections_tests.py
Comment (pbougue) [inline source/jormungandr/tests/line_sections_tests.py]: I think we should `assert not` and comment code here (with a note saying it doesn't currently work).
To me adding a test on something that's not supposed to happen just enforces it, and if that behaviour changes, we don't want the maintainer to revert it's code because of this test (it would actually be nice not to have that disruption).

But I don't understand why we get this disruption and not the other one...

File: source/jormungandr/tests/line_sections_tests.py
Comment (pbougue) [inline source/jormungandr/tests/line_sections_tests.py]: minor: "check`s`" or "check `that`"

File: source/jormungandr/tests/line_sections_tests.py
Comment (pbougue) [inline source/jormungandr/tests/line_sections_tests.py]: minor: "check`s`" or "check `that`"

**Review by pbougue — COMMENTED** [review]
Looks good to me except the one where we test that the behaviour is wrong.
(other minor remarks are very optional to process)

We may probably do some things super smart now with the new ptref filter from @TeXitoi ... But not in this PR!

**Comment by vpassama** [comment]
Ok @pbougue, I made the last changes, thank you for the review.

**Review by pbougue — APPROVED** [review]

**Comment by benoit-bst** [comment]
Jenkins, test this please !

**Pull request merged**
